### PR TITLE
[DD-456] Finalize Getting Started Experiment

### PR DESCRIPTION
**Pull request by tshen815**
[Ticket](https://circleci.atlassian.net/browse/DD-456)
[Preview](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/DD-456/finalize-getting-started-experiment-preview/)
[Optimizely](https://app.optimizely.com/v2/projects/16812830475/experiments/21268663100/variations)

⚠️ **NOTE**: we will merge this PR into #6544  and merge #6544 into master when we are ready to ship this experiment into prod at the conclusion of the Guided Language Tour experiments

# Changes

- add experiment layout
- import images needed for experiment 
- add styles needed for experiment
- implement experiment logic and variation segmentation for control and treatment groups
- bring together different experiment pieces listed in considerations

# Reasons
This experiment will aim to provide users with a revamped onboarding experience.

# Considerations 
As this experiment was broken down to multiple segments, this PR is unifying all the separate parts of the experiment for production: 

1.  #6544 
2.  #6564
3.  #6556
4.  #6567
5.  #6591
6. #6644

# Amplitude Events & Properties

1. clicked-landing-page-badge
- `badgeText`
- `badgeHref`

2. clicked-on-getting-started-guide-link
- `linkText`
- `linkHref`
- `linkIndexOnPage`
- `page`

# Demo 

**Desktop**

https://user-images.githubusercontent.com/57234605/162354734-14e76f4a-3149-4d7b-a103-7abf09f9b4b6.mp4

**Mobile**

https://user-images.githubusercontent.com/57234605/162354754-0d9cfdf2-c200-4446-a0f2-380c7d3dee4d.mp4

### CODE REVIEW

File: src/js/experiments/gettingStartedGuide.js
Comment (teesloane) [inline src/js/experiments/gettingStartedGuide.js:69]: Is the `/20` here because there are 20 links on the page? If so, could we maybe do `links.length` ?

**Review by teesloane — COMMENTED** [review]
Awesome work. Seeing all those `"capture"` tags makes me all the more glad we're moving to React. 

Thanks for the very detailed PR description with the links to other PRs! 

I suspect this is the final PR of changes - I would _maybe_ suggest that we merge it into the 450-base branch so that the "changed files" is a bit simpler in your PR and we can focus on reviewing just your changes - since everything else has already been merged into 450 (the badge, the two up, the styles etc) - we can then merge that into master being a bit more confident that everything there has already been reviewed to some degree.

File: src/js/experiments/gettingStartedGuide.js
Comment (tshen815) [reply src/js/experiments/gettingStartedGuide.js:69]: Yep! Great call out, im not sure what I was thinking with that string lol

**Comment by tshen815** [comment]
> Awesome work. Seeing all those `"capture"` tags makes me all the more glad we're moving to React.
> 
> Thanks for the very detailed PR description with the links to other PRs!
> 
> I suspect this is the final PR of changes - I would _maybe_ suggest that we merge it into the 450-base branch so that the "changed files" is a bit simpler in your PR and we can focus on reviewing just your changes - since everything else has already been merged into 450 (the badge, the two up, the styles etc) - we can then merge that into master being a bit more confident that everything there has already been reviewed to some degree.

Thats a great point, I totally agree! I was actually wondering how we could best to do and I think what you suggested would be the most logical way to get everything together.

File: src/js/experiments/gettingStartedGuide.js
Comment (teesloane) [inline src/js/experiments/gettingStartedGuide.js:54]: I recommend that we move this code and the code for tracking Links into a single function: `setupTracking`.

It's somewhat a preference thing - but imo,  it's easier to scan the bottom of the file in the treatment block and just see a list of named functions - this enables someone coming to read the code in the future to not have to scan each block of code to know what it does before realizing that it probably isn't what they need to change - vs - reading 4 lines of function names and (hopefully, if they are well named) jump to that function and make the changes they need to make.

**Review by teesloane — APPROVED** [review]
Looks excellent

I guess we just have to integrate the getting-started experiment for this to get in?

File: src/js/experiments/gettingStartedGuide.js
Comment (tshen815) [reply src/js/experiments/gettingStartedGuide.js]: great call out! I refactored the code for tracking and added in `setupTracking` at the bottom. I think once we merge this branch into [DD-450](https://github.com/circleci/circleci-docs/pull/6544) we should be good to merge that into master when we want to start running the experiment.

**Review by devinmarieb — APPROVED** [review]

**Review by Mayuko-Sa — APPROVED** [review]

**Review by chantelle11 — APPROVED** [review]
Great work on this!

**Review by teesloane — APPROVED** [review]

**Pull request merged**
